confluent-kafka-plugins/src/main/java/io/cdap/plugin/confluent/streaming/sink/ConfluentStreamingSinkConfig.java

```diff
@@ -207,14 +209,16 @@ public void validate(Schema inputSchema, FailureCollector collector) {
       ConfigValidations.validateBrokers(brokers, NAME_BROKERS, collector);
     }
 
-    if (!containsMacro(NAME_CLUSTER_API_KEY) && Strings.isNullOrEmpty(clusterApiKey)) {
-      collector.addFailure("Cluster API Key must be provided.", null)
-        .withConfigProperty(NAME_CLUSTER_API_KEY);
+    if (!containsMacro(NAME_CLUSTER_API_KEY) && !Strings.isNullOrEmpty(clusterApiKey) &&
+      Strings.isNullOrEmpty(clusterApiSecret)) {
+      collector.addFailure("Cluster API Secret should be provided when Cluster API Key is used.", null)
```

Comment:
Also check for macro in secret and test if key is given and secret is macro, is it working properly

Reply:
Added macro validation check for both Cluster API Key and Cluster API Secret. 8177e185ed4a426015592a0a227c63456179f85d